analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -115,8 +115,13 @@ public abstract class EmptyCollectionsShouldNotBeEnumeratedBase : SymbolicRuleCh
     protected override ProgramState PreProcessSimple(SymbolicContext context)
     {
         var operation = context.Operation.Instance;
-        if ((operation.AsObjectCreation()?.Type.IsAny(TrackedCollectionTypes) ?? false)
-            || IsEmptyArray(operation))
+        if (operation.AsObjectCreation() is { } objectCreation && objectCreation.Type.IsAny(TrackedCollectionTypes))
+        {
+            return ConstraintFromCollectionCreation(context.State, objectCreation) is { } constraint
```

Comment:
I'm not very happy with the name, as it's confusing `objectCreation` vs. the argument value. What about something like `CollectionArgumentConstraint`?